implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -80,6 +85,7 @@ public String getValue(String name) {
             return value;
         }
 
+        cache.put(name, NULL_VALUE);
         return null;
     }
 
```

Comment:
Something like this:

```suggestion
    Object writeReplace() {
        return new Ser();
    }

    static final class Ser implements Serializable {
        private static final long serialVersionUID = something;

        Object readResolve() {
            return new EnvConfigSource();
        }
    }
```